### PR TITLE
feat: add exercise_options_position

### PR DESCRIPTION
**Pull request by hiohiohio**
Context:
- A new API endpoint to exercise  an option contract (BETA) is introduced (https://docs.alpaca.markets/v1.1/reference/optionexercise)
- This PR support the API endpoint

Changes:
- add exercise_options_position

### CODE REVIEW

File: alpaca/common/utils.py [hunk 1]
```diff
@@ -54,6 +54,28 @@ def validate_symbol_or_asset_id(
...
+    symbol_or_contract_id: Union[UUID, str]
+) -> Union[UUID, str]:
+    """
+    A helper function to eliminate duplicate checks of symbols or contract id.
+
+    If the argument given is a string, assumed to be a symbol name. If a UUID object, assumed to be an contract id.
+    ValueError if neither type.
+
+    Args:
+        symbol_or_contractt_id: String representing a symbol name or a UUID representing an contract id.
```
Comment (Neal) [inline alpaca/common/utils.py:67]: typo:

```suggestion
        symbol_or_contract_id: String representing a symbol name or a UUID representing an contract id.
```

File: alpaca/trading/client.py [hunk 2]
```diff
@@ -317,6 +321,28 @@ def close_position(
 
         return Order(**response)
 
+    def exercise_option_contract(
```
Comment (Neal) [inline alpaca/trading/client.py:324]: `exercise_options_position` perhaps?

Comment (Neal) [reply alpaca/trading/client.py:324]: or maybe not as this is what we call it in the docs

File: alpaca/trading/client.py [hunk 3]
```diff
@@ -317,6 +321,28 @@ def close_position(
...
+        To cancel an exercise request or to submit a Do-not-exercise (DNE) instruction, please contact our support team.
+
+        Args:
+            symbol_or_contract_id (Union[UUID, str]): Option contract symbol or ID.
+
+        Returns:
+            None
+        """
+        symbol_or_contract_id = validate_symbol_or_contract_id(symbol_or_contract_id)
+        response = self.post(
```
Comment (matebudai) [inline alpaca/trading/client.py:342]: I believe `response` is unused. Not sure if you wanted to log or something.

Comment (hiohiohio) [reply alpaca/trading/client.py:342]: Good catch! Please let me remove it.

File: alpaca/common/utils.py [hunk 4]
```diff
@@ -54,6 +54,28 @@ def validate_symbol_or_asset_id(
...
+    Args:
+        symbol_or_contract_id: String representing a symbol name or a UUID representing an contract id.
+
+    Returns:
+        String if symbol, UUID if contract id.
+    """
+    if isinstance(symbol_or_contract_id, (UUID, str)):
+        return symbol_or_contract_id
+    raise ValueError(
+        f"symbol_or_contract_id must be a UUID of an contract id or a string of a symbol."
```
Comment (matebudai) [inline alpaca/common/utils.py:75]: f-string, but there are no interpolated variables

Comment (hiohiohio) [reply alpaca/common/utils.py]: Thank you! Removed.

File: alpaca/common/utils.py [hunk 5]
```diff
@@ -54,6 +54,28 @@ def validate_symbol_or_asset_id(
     )
 
 
+def validate_symbol_or_contract_id(
+    symbol_or_contract_id: Union[UUID, str]
+) -> Union[UUID, str]:
+    """
+    A helper function to eliminate duplicate checks of symbols or contract id.
+
+    If the argument given is a string, assumed to be a symbol name. If a UUID object, assumed to be an contract id.
```
Comment (matebudai) [inline alpaca/common/utils.py:63]: minor -> "a contract id'

Comment (hiohiohio) [reply alpaca/common/utils.py]: Thank you! Fixed!

**Review by matebudai — APPROVED** [review]

File: alpaca/trading/client.py (hunk 2 above)
Comment (hiohiohio) [reply alpaca/trading/client.py:324]: Thank you! agreed to rename!

**Pull request merged**